packages/react/dialog/src/dialog.tsx

```diff
@@ -516,10 +519,10 @@ For more information, see https://radix-ui.com/primitives/docs/components/${titl
 
   React.useEffect(() => {
     if (titleId) {
-      const hasTitle = document.getElementById(titleId);
+      const hasTitle = contentRef.current?.querySelector(`#${titleId}`);
```

Comment:
titleId needs to be sanitized in case it contains `:` - which is the case for react <19

```suggestion
      const hasTitle = contentRef.current?.querySelector(`#${titleId.replaceAll(':', '\\:')}`);
```

could also use `CSS.escape` instead of manual string replace

Reply:
Hello @radnan thx for review.
I found this PR was posted before, so I plan to close this PR and post your review on the previous PR.

Reply:
@chaance This bug is in high demand for a fix. What can I do?